tests/Unit/DataStructures/Test_DataVector.py

```diff
@@ -90,5 +99,267 @@ def test_output(self):
+        self.assertEqual(b.log()[1], math.log(0.5))
+
+    def test_log2(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.log2(), DataVector(5, 2.0))
+        b = DataVector(2, 1.0)
+        b[1] = 0.5
+        self.assertEqual(b.log2()[0], 0.0)
+        self.assertEqual(b.log2()[1], -1.0)
+
+    def test_log10(self):
+        a = DataVector(5, 100.0)
+        self.assertEqual(a.log10(), DataVector(5, 2.0))
+        b = DataVector(2, 1.0)
+        b[1] = 0.1
+        self.assertEqual(b.log10()[0], 0.0)
+        self.assertEqual(b.log10()[1], -1.0)
+
+    def test_max(self):
+        a = DataVector(5, 4.0)
+        a[1] = 5.0
+        self.assertEqual(a.max(), 5.0)
+        b = DataVector(2, 1.0)
+        b[1] = 0.1
+        self.assertEqual(b.max(), 1.0)
+
+    def test_min(self):
+        a = DataVector(5, 4.0)
+        a[1] = 2.0
+        self.assertEqual(a.min(), 2.0)
+        b = DataVector(2, 1.0)
+        b[1] = -0.5
+        self.assertEqual(b.min(),-0.5)
+
+    def test_pow(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.pow(2), DataVector(5, 16.0))
+        b = DataVector(2, 1.0)
+        b[1] = -0.5
+        self.assertEqual(b.pow(2)[0], 1.0)
+        self.assertEqual(b.pow(2)[1], 0.25)
+
+    def test_sin(self):
+        a = DataVector(5, 6.7)
+        self.assertEqual(a.sin(),DataVector(5, math.sin(6.7)))
+        b = DataVector(2, 0.0)
+        b[1] = -0.5
+        self.assertEqual(b.sin()[0], 0.0)
+        self.assertEqual(b.sin()[1], math.sin(-0.5))
+
+    def test_sinh(self):
+        a = DataVector(5, 3.0)
+        self.assertEqual(a.sinh(), DataVector(5, math.sinh(3.0)))
+        b = DataVector(2, 0.0)
+        b[1] = -0.5
+        self.assertEqual(b.sinh()[0], 0.0)
+        self.assertEqual(b.sinh()[1], math.sinh(-0.5))
+
+    def test_sqrt(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.sqrt(), DataVector(5, 2.0))
+        b = DataVector(2, 1.0)
+        b[1] = 0.25
+        self.assertEqual(b.sqrt()[0], 1.0)
+        self.assertEqual(b.sqrt()[1], 0.5)
+
+    def test_step_function(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.step_function(),DataVector(5, 1.0))
+        b = DataVector(2, 1.0)
+        b[1] = -0.5
+        self.assertEqual(b.step_function()[0], 1.0)
+        self.assertEqual(b.step_function()[1], 0.0)
+
+    def test_tan(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.tan(), DataVector(5, math.tan(4.0)))
+        b = DataVector(2, 0.0)
+        b[1] = -0.5
+        self.assertEqual(b.tan()[0], 0.0)
+        self.assertEqual(b.tan()[1], math.tan(-0.5))
+
+    def test_tanh(self):
+        a = DataVector(5, 4.0)
+        self.assertEqual(a.tanh(), DataVector(5, math.tanh(4.0)))
+        b = DataVector(2, 0.0)
+        b[1] = -0.5
+        self.assertEqual(b.tanh()[0], 0.0)
+        self.assertEqual(b.tanh()[1], math.tanh(-0.5))
+
+    #Implicitly testing the iterator
+    def test_numpy_compatibility(self):
```

Comment:
How does this test the iterator? It worked in my tests before `__iter__` was added. Better add another test for the iterator.

Reply:
In this small numpy compatability test you could also also add a test along the lines of `np.array([1, 2, 3]) + DataVector([0, 1, 2]) == np.array([1, 3, 5])`

Reply:
Could you please give an example of what you're looking for when you say "add another test for the iterator"? Thanks!

Reply:
```python
dv = DataVector([1, 2, 3])
for i, x in enumerate(dv):
    self.assertEqual(x, dv[i])
```
This would also fail if it terminated with an out-of-bounds exception as it did before.